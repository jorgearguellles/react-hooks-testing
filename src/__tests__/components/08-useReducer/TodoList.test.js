import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { demoTodos } from '../../fixtures/demoTodos';
import { TodoList } from '../../../components/08-useReducer/TodoList';


describe('Testing <TodoList />', ()=>{

  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow( 
    <TodoList 
      todos={demoTodos}
      handleDelete={ handleDelete }
      handleToggle={ handleToggle }
    />
  );

  test('1. SnapShot', ()=>{
    expect(wrapper).toMatchSnapshot();
  });

  test('2. Should to have two <TodoLitItem />', ()=>{
    expect( wrapper.find('TodoListItem').length ).toBe(2);
    expect( wrapper.find('TodoListItem').length ).toBe( demoTodos.length );
  });
  
  test('3. <TodoLitItem /> should receive props: handleDelete, handle Toggle', ()=>{
    const handleDeleteProp = wrapper.find('TodoListItem').at(0).prop('handleDelete');
    expect( typeof(handleDeleteProp) ).toBe('function');
    expect( handleDeleteProp ).toEqual( expect.any(Function) );

    const handleToggleProp = wrapper.find('TodoListItem').at(0).prop('handleToggle');
    expect( typeof(handleToggleProp) ).toBe('function');
    expect( handleToggleProp ).toEqual( expect.any(Function) );
  })
})