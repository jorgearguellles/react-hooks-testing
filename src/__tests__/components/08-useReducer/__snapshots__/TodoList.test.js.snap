// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TodoList /> 1. SnapShot 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Learn React.js",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Learn MongoDB",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
